streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -139,7 +140,7 @@ public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuil
         Objects.requireNonNull(materialized, "materialized can't be null");
         // explicitly disable logging for global stores
         materialized.withLoggingDisabled();
-        final StoreBuilder storeBuilder = new KeyValueStoreMaterializer<>(materialized).materialize();
+        final StoreBuilder storeBuilder = new KeyValueStoreMaterializer<>(materialized, this).materialize(KTableImpl.SOURCE_NAME);
```

Comment:
hmm... shouldn't we add `KTableImpl.STATE_STORE_NAME` also?

Reply:
EDIT: actually, how about the following: add a protected `storeName` setter in `Materialized`, and then we can just add the additional logic in this class by checking the `MaterializedInternal` object has either storeName or StoreSupplier, and if none is provided call the setter with the store name from `newStoreName(SOURCE_NAME)`.

Then we would not need any other classes' modifications.

Reply:
I'm not sure i follow

Reply:
I think it might be better to pass the `InternalNameProvider` and `prefix` into `MaterializedInternal` at construction as this will be required in other places. That way at least the logic will all be within the `MaterializedInternal` ctor